core/src/test/java/org/infinispan/conflict/resolution/ConflictResolutionManagerTest.java

```diff
@@ -0,0 +1,188 @@
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.Cache;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.container.entries.InternalCacheValue;
+import org.infinispan.context.Flag;
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.remoting.rpc.RpcManager;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.statetransfer.StateConsumer;
+import org.infinispan.statetransfer.StateRequestCommand;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.topology.CacheTopology;
+import org.infinispan.tx.dld.ControlledRpcManager;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+@Test(groups = "functional", testName = "conflict.resolution.ConflictResolutionManagerTest")
+public class ConflictResolutionManagerTest extends MultipleCacheManagersTest {
+
+   private static final int NUMBER_OF_OWNERS = 2;
+   private static final int NUMBER_OF_NODES = NUMBER_OF_OWNERS + 2;
+   private static final int NUMBER_OF_CACHE_ENTRIES = 100;
+   private static final int INCONSISTENT_VALUE_INCREMENT = 10;
+   private static final int NULL_VALUE_FREQUENCY = 20;
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      createClusteredCaches(NUMBER_OF_NODES, getConfigurationBuilder());
+   }
+
+   private ConfigurationBuilder getConfigurationBuilder() {
+      ConfigurationBuilder cacheConfigBuilder = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC);
+      cacheConfigBuilder.clustering()
+            .hash().numOwners(NUMBER_OF_OWNERS)
+            .stateTransfer().fetchInMemoryState(true);
+      return cacheConfigBuilder;
+   }
+
+   @BeforeMethod
+   public void populateCaches() {
+      AdvancedCache<Integer, String> cache0 = advancedCache(0);
+      IntStream.range(0, NUMBER_OF_CACHE_ENTRIES).forEach(i -> cache0.put(i, "v" + i));
+   }
+
+   @Test(expectedExceptions = IllegalStateException.class,
+         expectedExceptionsMessageRegExp = ".* Unable to retrieve conflicts for key .*")
+   public void testGetAllVersionsDuringStateTransfer() {
+      AdvancedCache cache0 = advancedCache(0);
+      StateConsumer sc = TestingUtil.extractComponent(cache0, StateConsumer.class);
+
+      sc = spy(sc);
+      doReturn(true).when(sc).isStateTransferInProgressForKey(any());
```

Comment:
IMO too synthetic test, you just check that a method is called, but it's not returning the actual value.